system/Superglobals.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter;
+
+/**
+ * Superglobals manipulation.
+ *
+ * @internal
+ */
+final class Superglobals
+{
+    private array $server;
+    private array $get;
+
+    public function __construct(?array $server = null, ?array $get = null)
+    {
+        $this->server = $server ?? $_SERVER;
+        $this->get    = $get ?? $_GET;
+    }
+
+    public function server(string $key): ?string
+    {
+        return $this->server[$key] ?? null;
+    }
+
+    public function setServer(string $key, string $value): void
+    {
+        $this->server[$key] = $value;
+        $_SERVER[$key]      = $value;
+    }
+
+    /**
+     * @return array|string|null
+     */
+    public function get(string $key)
+    {
+        return $this->get[$key] ?? null;
+    }
+
+    public function setGetArray(array $array): void
```

Comment:
Did you only add setters for a single SERVER key and the entire GET array because that is all the framework used? I could see wanting at least `setGet()` as well.

Reply:
Added.